specifyweb/frontend/js_src/lib/components/Forms/Save.tsx

```diff
@@ -204,17 +213,69 @@ export function SaveButton<SCHEMA extends AnySchema = AnySchema>({
       onClick={(): void => {
         // Scroll to the top of the form on clone
         smoothScroll(form, 0);
-        loading(handleClick().then(handleAdd));
+        loading(
+          handleClick()
+            .then((result) => {
+              const ids: WritableArray<number> = [];
+              if (originalResourceId !== undefined) {
+                ids.push(originalResourceId);
+              }
+              if (Array.isArray(result)) {
```

Comment:
change the signature of handleAdd() to make it be callable with an array of resources rather than one resource
and use that to simplify the code here
that would also shift the responsibility to the consumer of Save for deciding how it wants to handle or not to handle the case of more than one resource being present (i.e imagine that some code right now does a redirect on handleAdd() - by calling handleAdd multiples times here, that code is now potentially broken; where as by changing handleAdd signature to include array, that place would now be forced to consider how the array case needs to be handled)

Reply:
and once handleAdd() is being called with an array, there won't be need for handleCarryBulk() anymore as it can be combined

Reply:
Tagging you for a quick review to see if I'm on the right track with this

Reply:
looking better, but still work to do
could you tag me again once the TODOs you added are resolved?

Reply:
I'm not sure how to get rid of handleCarryBulk. I think I would still need to pass some sort of callback down from `RecordSet.tsx` to `ResourceView.tsx` to create a record set from an array of resources so that the consumer of `Save.tsx` (ResourceView) can define the behavior for when `resources.length > 1`?

Or I can define the behavior for `resources.length > 1` higher up the parent chain (maybe in `RecordSelectorFromIds` or `RecordSet` which could eliminate the need for the callback. What would be better?

Reply:
moved the bulk carry logic into onClone in RecordSet as that's the same callback being used for adding new resources